packages/vscode-extension/src/devices/AndroidEmulatorDevice.ts

```diff
@@ -371,6 +377,51 @@ export class AndroidEmulatorDevice extends DeviceBase {
     ]);
   }
 
+  async mirrorNativeLogs(build: AndroidBuildResult) {
+    const extractPidFromLogcat = async () =>
+      new Promise<string>(async (resolve, reject) => {
+        const regexString = `Start proc ([0-9]{4}):${build.packageName}`;
+        const process = exec(ADB_PATH, ["logcat", "-e", regexString]);
+        this.nativeLogsCancelToken.adapt(process);
+
+        lineReader(process).onLineRead((line) => {
+          const regex = new RegExp(regexString);
+
+          if (regex.test(line)) {
+            const groups = regex.exec(line);
+            const pid = groups?.[1];
+            process.kill();
+
+            if (pid) {
+              resolve(pid);
+            } else {
+              reject(new Error("PID not found"));
+            }
+          }
+        });
+
+        // We should be able to get pid immediately, if we're not getting it in 2s, then we reject to not block the app.
+        setTimeout(() => {
+          process.kill();
+          reject(new Error("Timeout while waiting for app to start to get the process PID."));
+        }, 2000);
```

Comment:
this could be too agressive timeout. This timeout doesn't cause any delays, it is just a safeguard to avoid the logcat process to run very long if things go wrong. Would be safer if we set it to something higher like 10seconds for example.